### ISSUE TITLE
show_diff  has been deprecated

### ISSUE THREAD

**Issue by herisanu** [saltstack-formulas/users-formula#132]
Let's change show_diff into show_changes: False

```
[WARNING ] /usr/lib/python2.7/site-packages/salt/states/file.py:1655: DeprecationWarning: The 'show_diff' argument to the file.managed state has been deprecated, please use 'show_changes' instead.
```

How to re-create: run formula on latest saltstack.
Link to code: https://github.com/saltstack-formulas/users-formula/blob/master/users/init.sls#L180 

Would you be ok with this change?

**Comment by blbradley** [comment]
I see `show_diff` in the 2016.3.3 docs. So, I don't think this is a good idea yet.

Also, backwards compatibility will need to be maintained for a long period of time.

**Comment by herisanu** [comment]
Yeah, backwards compatibility will be an issue. Would you consider using jinja for this?
Something along the lines of:

```
{{ users.show_diff }}: False
```

Where show_diff is set in defaults.yaml or map.jinja (with a grains filter on saltversion maybe)
This will give me the option of throwing this in a pillar.

**Comment by blbradley** [comment]
I would use a conditional that checks the salt version and uses the appropriate parameter. No pillar config required.

**Comment by herisanu** [comment]
Yeah, right. Or that. Is it cool to send a PR with this?

**Comment by blbradley** [comment]
I believe you should wait until there is a stable version of Salt that shows up in the docs with `show_changes`.

But don't mind me. I don't actually use this formula.

**Comment by noelmcloughlin** [comment]
@herisanu Now is good time to do this. thx

**Comment by herisanu** [comment]
Heh an issue from 2016 :). Let me re-read the formula and create a PR then.

**Comment by noelmcloughlin** [comment]
Chuckle, out past comes back to haunt us sometimes
